uniffi_bindgen/src/bindings/kotlin/templates/CallbackInterfaceTemplate.kt

```diff
@@ -0,0 +1,75 @@
+{% let type_name = obj.name()|class_name_kt %}
+public interface {{ type_name }} {
+    {% for meth in obj.methods() -%}
+    fun {{ meth.name()|fn_name_kt }}({% call kt::arg_list_decl(meth) %})
+    {%- match meth.return_type() -%}
+    {%- when Some with (return_type) %}: {{ return_type|type_kt -}}
+    {%- else -%}
+    {%- endmatch %}
+    {% endfor %}
+}
+
+{% let canonical_type_name = format!("CallbackInterface{}", type_name) %}
+{% let callback_internals = format!("{}Internals", canonical_type_name) -%}
+{% let callback_interface_impl = format!("{}FFI", canonical_type_name) -%}
+
+internal class {{ callback_interface_impl }} : ForeignCallback {
+    @Suppress("TooGenericExceptionCaught")
+    override fun invoke(handle: Long, method: Int, args: RustBuffer.ByValue): RustBuffer.ByValue {
+        return {{ callback_internals }}.handleMap.callWithResult(handle) { cb -> 
+            when (method) {
+                0 -> {{ callback_internals }}.drop(handle)
+                {% for meth in obj.methods() -%}
+                {% let method_name = format!("invoke_{}", meth.name())|fn_name_kt -%}
+                {{ loop.index }} -> this.{{ method_name }}(cb, args)
+                {% endfor %}
+                else -> RustBuffer.ByValue()
+            }
+        }
+    }
+
+    {% for meth in obj.methods() -%}
+    {% let method_name = format!("invoke_{}", meth.name())|fn_name_kt %}
+    private fun {{ method_name }}(kotlinCallbackInterface: {{ type_name }}, args: RustBuffer.ByValue): RustBuffer.ByValue =
+        try {
+        {#- Unpacking args from the RustBuffer #}
+            {%- if meth.arguments().len() != 0 -%}
+            {#- Calling the concrete callback object #}
+            val buf = args.asByteBuffer() ?: throw InternalError("No ByteBuffer in RustBuffer; this is a Uniffi bug")
```

Comment:
What will actually happen to this exception if it does get thrown? Or equivalently, what will happen if the user's provided implementation method throws?

The JNA docs say:

> A callback should generally never throw an exception, since it doesn't necessarily have an encompassing
> Java environment to catch it. Any exceptions thrown will be passed to the default callback exception handler.

But it's not at all obvious to me what that actually means, or how the rust code will respond if allowed to continue running after this happens.

Should we try to catch exceptions and signal the error to the Rust code, maybe using an `out ExternError` argument like we do for calls from the foreign language into Rust?